src/functionality.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-use-before-define */
+const taskList = document.getElementById('task-list');
+
+let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+export function addTask(description) {
+  const task = {
+    description,
+    completed: false,
+    index: tasks.length,
+  };
+  tasks.push(task);
+  renderTask(task);
+  updateIndex();
+  saveTasks();
+}
+
+export function renderTask(task) {
+  const li = document.createElement('li');
+  li.innerHTML = `
+    <input type="checkbox" class="checkbox" ${task.completed ? 'checked' : ''}>
+    <span class="task-description ${task.completed ? 'completed' : ''}">${
+  task.description
+}</span>
+    <span class="delete-task">X</span>
+  `;
+  li.classList.add('task');
+  li.index = task.index;
+  taskList.appendChild(li);
+
+  const checkbox = li.querySelector('.checkbox');
```

Comment:
* As stated in the [project requiremen](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20editing%20task%20descriptions.)t you should Implement a function for editing task descriptions, this functionality is missing kindly add a function that makes this work as currently, i can edit a task description